apps/financial-aid/web-veita/src/routes/ApplicationsOverview/applicationsOverviewProcessed.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useEffect, useState } from 'react'
+import { useRouter } from 'next/router'
+import { useLazyQuery } from '@apollo/client'
+
+import { Text, Box, Pagination } from '@island.is/island-ui/core'
+import {
+  ApplicationsTable,
+  FilterPopover,
+} from '@island.is/financial-aid-web/veita/src/components'
+import {
+  ApplicationState,
+  ApplicationPagination,
+  applicationPageSize,
+} from '@island.is/financial-aid/shared/lib'
+import { ApplicationFilterQuery } from '@island.is/financial-aid-web/veita/graphql/sharedGql'
+import { navigationItems } from '@island.is/financial-aid-web/veita/src/utils/navigation'
+import { container } from './applicationsOverviewProcessed.css'
+
+interface Filters {
+  selectedStates: ApplicationState[]
+  selectedMonths: number[]
+}
+
+export const ApplicationsOverviewProcessed = () => {
+  const router = useRouter()
+
+  const [currentPage, setCurrentPage] = useState<number>(
+    router?.query?.page ? parseInt(router.query.page as string) : 1,
+  )
+  const [filters, setFilters] = useState<Filters>({
+    selectedStates: router?.query?.state
+      ? ((router?.query?.state as string).split(',') as ApplicationState[])
+      : [],
+    selectedMonths: router?.query?.month
+      ? (router?.query?.month as string).split(',').map(Number)
+      : [],
+  })
+
+  const onChecked = (item: ApplicationState | number, checked: boolean) => {
+    const filtersCopy = { ...filters }
+
+    if (typeof item === 'number') {
+      checked
+        ? filters.selectedMonths.push(item)
+        : filters.selectedMonths.splice(filters.selectedMonths.indexOf(item), 1)
+    } else {
+      checked
+        ? filters.selectedStates.push(item)
+        : filters.selectedStates.splice(filters.selectedStates.indexOf(item), 1)
+    }
+
+    setFilters(filtersCopy)
+  }
+
+  const onFilterClear = () => {
+    setFilters({ selectedMonths: [], selectedStates: [] })
+    setCurrentPage(1)
+    filter(1, { selectedMonths: [], selectedStates: [] })
```

Comment:
Please use the `useEffect` hook with dependencies to the state to trigger the filtering.